packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -321,49 +346,61 @@ Future<XcodeBuildResult> buildXcodeProject({
     );
   }
 
-  final Status status = logger.startProgress('Running Xcode build...', expectSlowOperation: true);
-  final RunResult result = await runAsync(
-    commands,
+  final Status buildStatus =
+      logger.startProgress('Running Xcode build...', expectSlowOperation: true);
+  final RunResult buildResult = await runAsync(
+    buildCommands,
     workingDirectory: app.appDirectory,
     allowReentrantFlutter: true
   );
-  status.stop();
-  if (result.exitCode != 0) {
+  buildStatus.stop();
+
+  // Run -showBuildSettings again but with the exact same parameters as the build.
+  final Map<String, String> buildSettings = parseXcodeBuildSettings(runCheckedSync(
+    buildCommands..add('-showBuildSettings'),
```

Comment:
I suggest we pass in a new list with the added flag rather than mutate `buildCommands`. It is used later for diagnostics and I believe it should reflect the actual command line that produced the `buildResult` above.

Just to avoid any surprises down the road.

Reply:
Oh ya, totally good point. Thanks.